cpp/log/tree_signer-inl.h

```diff
@@ -242,32 +242,36 @@ typename TreeSigner<Logged>::UpdateResult TreeSigner<Logged>::UpdateTree() {
     return INSUFFICIENT_DATA;
   }
 
-  VLOG(1) << "Building tree starting from " << next_seq;
-  while (true) {
-    EntryHandle<Logged> logged;
-    util::Status status(
-        consistent_store_->GetSequencedEntry(next_seq, &logged));
-    if (status.CanonicalCode() == util::error::NOT_FOUND) {
-      // no more certs to integrate (or a gap in the sequence numbers, but
-      // that'd be bad so we should bail anyway.)
-      break;
-    } else if (!status.ok()) {
-      LOG(WARNING) << "problem getting sequenced entry " << next_seq << ": "
-                   << status;
-      return DB_ERROR;
-    }
+  // TODO(alcutter): We probably should just be appending the entries we have
+  // locally and allow the followers to fetch the latest entries via the master
+  // node.
+  std::vector<EntryHandle<Logged>> sequenced_entries;
+  util::Status status(
+      consistent_store_->GetSequencedEntries(&sequenced_entries));
+  if (!status.ok()) {
+    LOG(WARNING) << "Failed to GetSequencedEntries: " << status;
+    return DB_ERROR;
+  }
+  typename std::vector<EntryHandle<Logged>>::const_iterator it(
+      sequenced_entries.begin());
+  while (it != sequenced_entries.end() &&
+         it->Entry().sequence_number() < next_seq) {
+    ++it;
+  }
 
+  VLOG(1) << "Building tree starting from " << next_seq;
+  for (; it != sequenced_entries.end(); ++it) {
     // Paranoid much?
-    CHECK(logged.Entry().has_sequence_number());
-    CHECK_EQ(next_seq, logged.Entry().sequence_number());
+    CHECK(it->Entry().has_sequence_number());
+    CHECK_EQ(next_seq, it->Entry().sequence_number());
 
-    if (!Append(logged.Entry())) {
+    if (!Append(it->Entry())) {
       LOG(ERROR) << "Assigning sequence number failed";
```

Comment:
Ah, I guess this is a message from the olden-days, when Appending implied sequencing.
I've updated it.